test/example_client_connection.erl

```diff
@@ -152,17 +153,19 @@ peer_needs_streams(C, bidi_streams, S) ->
 handle_info({'EXIT', _Pid, _Reason}, State) ->
     {ok, State};
 handle_info({quic, Sig, Stream, _} = Msg, #{streams := Streams} = S) when
-    %% @FIXME, not desired behavior.
-    %% Casued by inflight quic Msg during stream handoff
     Sig == peer_send_shutdown orelse Sig == stream_closed
 ->
-    {OwnerPid, Stream} = lists:keyfind(Stream, 2, Streams),
-    NewS =
-        case OwnerPid == owner_down orelse OwnerPid == closed of
-            true ->
-                quicer:async_shutdown_stream(Stream),
-                S#{streams := lists:keydelete(Stream, 2, Streams)};
-            false ->
-                error(fixme)
-        end,
-    {ok, S}.
+    case lists:keyfind(Stream, 2, Streams) of
+        {Reason, Stream} when
+            Reason =:= owner_down orelse
+                Reason =:= closed orelse
+                Reason =:= start_error
+        ->
+            _ = quicer:async_shutdown_stream(Stream),
+            {ok, S#{streams := lists:keydelete(Stream, 2, Streams)}};
+        {OwnerPid, Stream} when is_pid(OwnerPid) ->
+            {error, {fixme, bug_handoff_fail}};
```

Comment:
Supposed to fix in this PR?  Otherwise might be useful as a reference to add comment about this fixme.